apps/address-service/domains/common/utils/services/search/plugins/SearchByProvider.js

```diff
@@ -0,0 +1,78 @@
+const { AbstractSearchPlugin } = require('@address-service/domains/common/utils/services/search/AbstractSearchPlugin')
+const { Address } = require('@address-service/domains/address/utils/serverSchema')
+const { generateAddressKey } = require('@address-service/domains/common/utils/addressKeyUtils')
+const { getSearchProvider } = require('@address-service/domains/common/utils/services/providerDetectors')
+const get = require('lodash/get')
+
+class SearchByProvider extends AbstractSearchPlugin {
+
+    /**
+     * @param {String} s
+     * @returns {Promise<?Object>}
+     */
+    async search (s) {
+        const searchProvider = getSearchProvider(this.geo)
+        const godContext = this.keystoneContext.sudo()
```

Comment:
Could you please make names consistent.
```suggestion
        const sudoContext = this.keystoneContext.sudo()
```

Reply:
I'd like to show that this is a dangerous context. Any other developer will pay attention to it.

Reply:
Maybe `sudoContext` to avoid creating many one-off words…

Reply:
Windows users won't understand :)